core/src/main/scala/org/apache/spark/HttpServer.scala

```diff
@@ -155,6 +156,7 @@ private[spark] class HttpServer(
       throw new ServerStateException("Server is already stopped")
     } else {
       server.stop()
+      Option(server.getThreadPool).collect { case x: LifeCycle => x }.foreach(_.stop())
```

Comment:
I agree that when there are multiple chaining with not so obvious function/partial function applications, scala can become unreadable. However, for the case we are having here, I do think that using Option collect foreach is quite straightforward and easily readable (and compare to someone chaining multiple RDD operations, this is by far very readable).